flycheck.el

```diff
@@ -6920,6 +6921,62 @@ See URL `http://www.erlang.org/'."
    (error line-start (file-name) ":" line ": " (message) line-end))
   :modes erlang-mode)
 
+(defun contains-rebar-config (dir-name)
+  "Return DIR-NAME if DIR-NAME/rebar.config exists, nil otherwise."
+  (if (file-exists-p (expand-file-name "rebar.config" dir-name))
+      dir-name nil))
+
+(defun locate-rebar3-project-root (file-name &optional prev-file-name acc)
```

Comment:
Yes. You might want to explore how a dependency works, but that buffer might think it's the main project without this. Then it would try and fetch the dependencies dependencies, which have already been fetched by the root project.